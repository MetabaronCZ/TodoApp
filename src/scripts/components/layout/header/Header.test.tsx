import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { withMockedProviders } from 'test/component';
import { Header } from 'components/layout/header/Header';

describe('components/layout/Header', () => {
  it('should render correctly', () => {
    const { container } = render(withMockedProviders(<Header />));
    expect(container).toMatchSnapshot();
    expect(container.querySelector('[href="/settings"]')).toBeInTheDocument();
    expect(
      container.querySelector('[href="/folder/list"]'),
    ).toBeInTheDocument();
  });
});
