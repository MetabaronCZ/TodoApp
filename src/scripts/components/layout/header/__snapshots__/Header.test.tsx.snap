// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/layout/Header > should render correctly 1`] = `
.c5 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  width: 20px;
  height: 20px;
  line-height: 20px;
  font-style: normal;
  text-align: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  padding: 5px 10px;
  padding-left: 5px;
  padding-right: 5px;
  background: #f0f0f0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover:not(:disabled) {
  background-color: #e0e0e0;
}

.c4:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c4:disabled {
  color: #c0c0c0;
}

.c1 {
  font-family: Open Sans,Arial,sans-serif;
  color: #222;
  font-size: 14px;
  line-height: 20px;
  display: block;
  padding: 0 10px;
  font-size: 20px;
  line-height: 40px;
  font-weight: bold;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:focus {
  outline: 1px dotted #222;
  outline-offset: -1px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 0;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
}

<div>
  <header
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      ☑ 
      TODOLIST
    </a>
    <div
      class="c2"
    />
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/folder/list"
        title="Složky"
      >
        <i
          class="c5"
        >
          ★
        </i>
      </a>
      <a
        class="c4"
        href="/settings"
        title="Nastavení"
      >
        <i
          class="c5"
        >
          ✎
        </i>
      </a>
    </div>
  </header>
</div>
`;
